src/cargo/ops/cargo_remove.rs

```diff
@@ -59,7 +85,46 @@ pub fn remove(options: &RemoveOptions<'_>) -> CargoResult<()> {
             .warn("aborting remove due to dry run")?;
     } else {
         manifest.write()?;
+
+        if let Some(workspace_manifest) = workspace_manifest {
+            cargo_util::paths::write(
+                workspace.root_manifest(),
+                workspace_manifest.to_string().as_bytes(),
+            )?;
+        }
     }
 
     Ok(())
 }
+
+/// Get whether or not a dependency is marked as `workspace`.
+fn dependency_is_workspace(dep: &str, dep_table: &[String], manifest: &LocalManifest) -> bool {
+    if let Ok(toml_edit::Item::Table(table)) = manifest.get_table(dep_table) {
+        let value = table.get(dep).and_then(|i| i.get("workspace"));
+        if let Some(toml_edit::Item::Value(value)) = value {
+            return value.as_bool() == Some(true);
+        }
+    }
+
+    false
+}
+
+/// Get whether or not a dependency is depended upon in a workspace.
+fn dependency_in_workspace(dep: &str, exclude: &Package, workspace: &Workspace<'_>) -> bool {
+    workspace.members().filter(|p| *p != exclude).any(|p| {
+        p.dependencies()
+            .iter()
+            .any(|d| d.package_name().as_str() == dep)
+    })
+}
+
+/// Remove a dependency from a virtual workspace.
+fn remove_dependency_from_workspace(dep: &str, virtual_manifest: &mut toml_edit::Document) {
+    if let Some(toml_edit::Item::Table(table)) = virtual_manifest
+        .get_mut("workspace")
+        .and_then(|t| t.get_mut("dependencies"))
+    {
+        table.set_implicit(true);
+        table.remove(dep);
+    }
+}
```

Comment:
This mimics some of the work done in `cargo::util::toml_mut::manifest`, which explicitly does not support virtual manifests. Alternatively, we could expand `toml_mut` in some way to support this.